server/game/core/card/LeaderUnitCard.ts

```diff
@@ -1,39 +1,104 @@
 import Player from '../Player';
 import { LeaderCard } from './LeaderCard';
 import { InitiateAttackAction } from '../../actions/InitiateAttackAction';
-import { CardType, Location } from '../Constants';
+import { CardType, Location, LocationFilter } from '../Constants';
 import { WithCost } from './propertyMixins/Cost';
 import { WithUnitProperties } from './propertyMixins/UnitProperties';
+import type { UnitCard } from './CardTypes';
+import * as EnumHelpers from '../utils/EnumHelpers';
+import { IActionAbilityProps, IConstantAbilityProps, IReplacementEffectAbilityProps, ITriggeredAbilityProps } from '../../Interfaces';
+import * as Helpers from '../utils/Helpers';
+import AbilityHelper from '../../AbilityHelper';
+import Contract from '../utils/Contract';
 
 const LeaderUnitCardParent = WithUnitProperties(WithCost(LeaderCard));
 
-// TODO LEADERS: add custom defeat logic
 export class LeaderUnitCard extends LeaderUnitCardParent {
-    private _isDeployed = false;
-
-    public get isDeployed() {
-        return this._isDeployed;
-    }
-
     public override get type() {
-        return this._isDeployed ? CardType.LeaderUnit : CardType.Leader;
+        return this._deployed ? CardType.LeaderUnit : CardType.Leader;
```

Comment:
Not a must-have, but I think I prefer the "isDeployed" naming convention. It makes it clear without needing code hints that this is a boolean field

Reply:
agreed here

Reply:
As discussed in chat I'm good with this.